### PR TITLE
Updated some more XML comments

### PR DESCRIPTION
**Pull request by d3fkn1ght**
﻿## Description of Change
Updated XML Documenation for the following files:

Prism.Forms
ParameterExtension.cs

Prism.Core
ParameterBase.cs
ParametersExtensions.cs

ErrorsContainer.cs

IDestructible.cs

Prism.Wpf
IRegionNavigationJournal.cs

### Bugs Fixed

None

### API Changes

None

### Behavioral Changes

None

### PR Checklist

- [ ] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

File: src/Prism.Core/Common/ParametersExtensions.cs [hunk 1]
```diff
@@ -108,6 +139,12 @@ private static bool TryGetValueInternal(KeyValuePair<string, object> kvp, Type t
             return success;
         }
 
+        /// <summary>
+        /// Checks to see if key exists in IEnumberable of KeyValuePairs
```
Comment (dansiegel) [inline src/Prism.Core/Common/ParametersExtensions.cs:143]: Let's not reference the KeyValuePairs

File: src/Prism.Core/Mvvm/ErrorsContainer.cs [hunk 2]
```diff
@@ -12,7 +12,14 @@ namespace Prism.Mvvm
     public class ErrorsContainer<T>
     {
         private static readonly T[] noErrors = new T[0];
+        /// <summary>
```
Comment (dansiegel) [inline src/Prism.Core/Mvvm/ErrorsContainer.cs:15]: put a space between the xml docs and the previous line of code

File: src/Prism.Core/Mvvm/ErrorsContainer.cs [hunk 3]
```diff
@@ -12,7 +12,14 @@ namespace Prism.Mvvm
...
+        /// Delegate to be called when raiseErrorsChanged is invoked
+        /// </summary>
         protected readonly Action<string> raiseErrorsChanged;
+
+        /// <summary>
+        /// KeyValuePairs of the string error and a List of the types of the error object(s)
```
Comment (dansiegel) [inline src/Prism.Core/Mvvm/ErrorsContainer.cs:21]: This shouldn't reference the KeyValuePairs, it should reference the `<see cref="Dictionary{string, List{T}}" />`

File: src/Prism.Core/Navigation/IDestructible.cs [hunk 4]
```diff
@@ -1,7 +1,13 @@
 ﻿namespace Prism.Navigation
 {
+    /// <summary>
+    /// Interface for destuctible objects 
+    /// </summary>
     public interface IDestructible
     {
+        /// <summary>
+        /// Classic destructor for object to be called on destruction
+        /// </summary>
```
Comment (dansiegel) [inline src/Prism.Core/Navigation/IDestructible.cs:10]: This provides a spot to cleanup your View/ViewModel for any resources it has.

File: src/Forms/Prism.Forms/Xaml/ParameterExtension.cs [hunk 5]
```diff
@@ -3,11 +3,24 @@
...
+    /// <summary>
+    /// Extension methods for INavigation and IDialog parameters
+    /// </summary>
     public class ParameterExtension : Parameter, IMarkupExtension<Parameter>
     {
+        /// <summary>
+        /// Method to retrieve value for parameter
+        /// </summary>
+        /// <param name="serviceProvider">The service that supplies the parameters value</param>
+        /// <returns>This class</returns>
```
Comment (dansiegel) [inline src/Forms/Prism.Forms/Xaml/ParameterExtension.cs:15]: It returns the instance

File: src/Prism.Core/Common/ParametersBase.cs [hunk 6]
```diff
@@ -102,9 +102,9 @@ protected ParametersBase(string query)
             _entries.Select(x => x.Key);
 
         /// <summary>
-        /// Adds the key and value to the KeyValuePair<string,object> collection
+        /// Adds the key and value to the KeyValuePair&lt;string,object> collection
         /// </summary>
-        /// <param name="key">The key to reference this value in the KeyValuePair<string, object></param>
+        /// <param name="key">The key to reference this value in the KeyValuePair&lt;string, object></param>
```
[the remark this answers is not in the document]
Comment (d3fkn1ght) [reply src/Prism.Core/Common/ParametersBase.cs]: I have removed the keyvalue pair text from the doc comments

File: src/Prism.Core/Common/ParametersExtensions.cs [hunk 7]
```diff
@@ -5,13 +5,31 @@
...
     [EditorBrowsable(EditorBrowsableState.Never)]
     public static class ParametersExtensions
     {
+        /// <summary>
+        /// Searches <paramref name="parameters"/> for <paramref name="key"/>
+        /// </summary>
+        /// <typeparam name="T">The type of the parameter to return</typeparam>
+        /// <param name="parameters">A collection of parameters to search</param>
+        /// <param name="key">The key of the parameter to find</param>
+        /// <returns>A matching value of <langword>T</langword> if it exists</returns>
```
[the remark this answers is not in the document]
Comment (d3fkn1ght) [reply src/Prism.Core/Common/ParametersExtensions.cs:20]: I have changed the <langword> tag to a <typeparam> tag as requested

File: src/Prism.Core/Common/ParametersExtensions.cs [hunk 8]
```diff
@@ -48,6 +73,12 @@ public static bool TryGetValue<T>(this IEnumerable<KeyValuePair<string, object>>
             return false;
         }
 
+        /// <summary>
+        /// Searches <paramref name="parameters"/> for value referenced by <paramref name="key"/>
+        /// </summary>
+        /// <param name="parameters">A collection of parameters to search</param>
+        /// <param name="key">The key of the parameter to find</param>
+        /// <returns>An IEnumberable&lt;T> of all the values referenced by key</returns>
```
[the remark this answers is not in the document]
Comment (d3fkn1ght) [reply src/Prism.Core/Common/ParametersExtensions.cs:81]: I have changed the IEnumberable&lt;T> to IEnumerable{T}

File: src/Prism.Core/Common/ParametersExtensions.cs (hunk 1 above)
Comment (d3fkn1ght) [reply src/Prism.Core/Common/ParametersExtensions.cs:143]: I removed IEnumberable of KeyValuePairs and replaced it with parameter collection

File: src/Forms/Prism.Forms/Xaml/ParameterExtension.cs (hunk 5 above)
Comment (d3fkn1ght) [reply src/Forms/Prism.Forms/Xaml/ParameterExtension.cs]: I have changed the <returns> tag

File: src/Prism.Core/Mvvm/ErrorsContainer.cs (hunk 2 above)
Comment (d3fkn1ght) [reply src/Prism.Core/Mvvm/ErrorsContainer.cs:16]: Space added

File: src/Prism.Core/Mvvm/ErrorsContainer.cs (hunk 3 above)
Comment (d3fkn1ght) [reply src/Prism.Core/Mvvm/ErrorsContainer.cs:22]: I have replaced KeyValuePairs with <see cref="Dictionary{string, List{T}}" />

File: src/Prism.Core/Navigation/IDestructible.cs (hunk 4 above)
Comment (d3fkn1ght) [reply src/Prism.Core/Navigation/IDestructible.cs]: Modified comments to reflect better description

File: src/Forms/Prism.Forms/Xaml/ParameterExtension.cs [hunk 9]
```diff
@@ -3,11 +3,24 @@
 
 namespace Prism.Xaml
 {
+    /// <summary>
+    /// Extension methods for INavigation and IDialog parameters
```
Comment (dansiegel) [inline src/Forms/Prism.Forms/Xaml/ParameterExtension.cs:7]: These are not extensions methods. This is a XAML Extension

**Pull request merged**